scanner/src/main/kotlin/storages/CompositeStorage.kt

```diff
@@ -67,6 +68,33 @@ class CompositeStorage(
     override fun readInternal(pkg: Package, scannerCriteria: ScannerCriteria): Result<ScanResultContainer> =
         fetchReadResult(pkg.id) { read(pkg, scannerCriteria) }
 
+    override fun readInternal(
+        packages: List<Package>,
+        scannerCriteria: ScannerCriteria
+    ): Result<Map<Identifier, List<ScanResult>>> {
+        if (readers.isEmpty()) return Success(emptyMap())
+
+        val remainingPackages = packages.toMutableList()
+        val result = mutableMapOf<Identifier, List<ScanResult>>()
+        val failures = mutableListOf<String>()
+
+        readers.forEach { reader ->
+            when (val readerResult = reader.read(remainingPackages, scannerCriteria)) {
```

Comment:
Should the loop terminate as soon as `remainingPackages` is empty? I think, at least the Postgres storage will still execute a query even if the collection with packages is empty.

Reply:
Done.